demo/src/androidTest/java/com/braintreepayments/browserswitch/demo/DemoActivityTest.java

```diff
@@ -0,0 +1,54 @@
+package com.braintreepayments.browserswitch.demo;
+
+import android.content.Context;
+import android.content.Intent;
+
+import androidx.test.core.app.ApplicationProvider;
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+import androidx.test.platform.app.InstrumentationRegistry;
+import androidx.test.uiautomator.By;
+import androidx.test.uiautomator.UiDevice;
+import androidx.test.uiautomator.UiObjectNotFoundException;
+import androidx.test.uiautomator.UiSelector;
+import androidx.test.uiautomator.Until;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static org.junit.Assert.assertTrue;
+
+
+@RunWith(AndroidJUnit4.class)
+public class DemoActivityTest {
+    private static final long APP_LAUNCH_TIMEOUT_MS = 5000;
+    private static final String RETURN_URL_PREFIX = "com.braintreepayments.browserswitch.demo.browserswitch://";
+
+    @Test
+    public void testDemo() throws UiObjectNotFoundException {
+        UiDevice device = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation());
+        Context context = ApplicationProvider.getApplicationContext();
+
+        // Launch our app using UIAutomator instead of Espresso.
+        // If we use Espresso, our activity is killed when the browser opens.
+        Intent intent = context.getPackageManager().getLaunchIntentForPackage(BuildConfig.APPLICATION_ID);
+        context.startActivity(intent);
+        device.wait(Until.hasObject(By.pkg(BuildConfig.APPLICATION_ID).depth(0)), APP_LAUNCH_TIMEOUT_MS);
+
+        performBrowserSwitch(device, "Red", RETURN_URL_PREFIX + "?color=red");
+        performBrowserSwitch(device, "Green", RETURN_URL_PREFIX + "?color=green");
+        performBrowserSwitch(device, "Blue", RETURN_URL_PREFIX + "?color=blue");
+        performBrowserSwitch(device, "I don't like any of these colors",
+                RETURN_URL_PREFIX + "cancel?color=undefined");
+    }
+
+    private void performBrowserSwitch(UiDevice device,
+                                      String colorActionLabel,
+                                      String expectedReturnUrl) throws UiObjectNotFoundException {
+        UiSelector selector = new UiSelector();
+        device.findObject(selector.text("Start Browser Switch")).click();
```

Comment:
It probably would have been better to use a string resource id instead of `"Start Browser Switch"`. And perhaps also some string resource ids with formatting for the result and return texts. But:
* We have to use String literals anyway for interacting with the browser
* String resource ids aren't currently used in this demo, and I didn't want to change the app code just to make an instrumentation test suggestion

Reply:
I'm OK with having the raw string here, but it's only working for me if this string is in all caps: `"START BROWSER SWITCH"`.

Reply:
I see that on some versions of Android, the OS capitalizes the button text. I fixed this to do the look up by id instead.

However, now that I started to test on different android versions, on some versions (example: marshmallow) it doesn't seem to be able to find the links like "Red". I didn't have time yet to investigate this 😞